liblangutil/SourceReferenceExtractor.h

```diff
@@ -55,13 +56,18 @@ namespace SourceReferenceExtractor
 	struct Message
 	{
 		SourceReference primary;
-		std::string severity; // "Error", "Warning", "Info", ...
+		std::variant<Error::Type, Error::Severity> _typeOrSeverity;
 		std::vector<SourceReference> secondary;
 		std::optional<ErrorId> errorId;
 	};
 
-	Message extract(CharStreamProvider const& _charStreamProvider, util::Exception const& _exception, std::string _severity);
+	Message extract(
+		CharStreamProvider const& _charStreamProvider,
+		util::Exception const& _exception,
+		std::variant<Error::Type, Error::Severity> _typeOrSeverity
+	);
 	Message extract(CharStreamProvider const& _charStreamProvider, Error const& _error);
+	Message extract(CharStreamProvider const& _charStreamProvider, Error::Severity _severity, Error const& _error);
```

Comment:
Why not make it the last arg so that `_error` is the second one in both cases? Seems more consistent like this.
```suggestion
	Message extract(CharStreamProvider const& _charStreamProvider, Error const& _error, Error::Severity _severity);
```

Reply:
Ahh, I was trying to make it more so that the `_error` is the last argument in both.

Also, there seems to be a problem with putting `_error` as second.

We have 
`extract (_charStreamProvider, _error, _severity)`
and 
`extract(_charStreamProvider, _error, _typeOrSeverity)`

When I put `_severity` as the second argument, things work fine. But in the order as specified here, I start getting seg faults.  I suspect the function overloading here results in the wrong function being called.

Reply:
What if you make them both accept `_typeOrSeverity` as the last argument?

From what I can see the main difference is that one takes `util::Exception`, which does not include the error ID, and the other takes `langutil::Error` and is really just a convenience wrapper that adds the ID. It really looks like all the other args in them should be the same.

Reply:
I mean, doing that reduces the function to 
```
SourceReferenceExtractor::Message SourceReferenceExtractor::extract(
	CharStreamProvider const& _charStreamProvider,
	Error const& _error,
	std::variant<Error::Type, Error::Severity> _typeOrSeverity
)
{
	solAssert(std::holds_alternative<Error::Severity>(_typeOrSeverity));
	Message message = extract(_charStreamProvider, _error, std::get<Error::Severity>(_typeOrSeverity));
	message.errorId = _error.errorId();
	return message;
}
```
Here, the function will be calling itself and creating an infinite recursion. 
The other variant of this function that uses `_exception` actually implements the creation of a `Message` object. I'll be emulating that code for this function too.

Reply:
What if you cast `_error` in the call to `util::Exception`?

Reply:
Yep, doing that works. 

Have done that and removed the redundant code.